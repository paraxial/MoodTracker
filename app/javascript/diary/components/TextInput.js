import React from 'react';
import PropTypes from 'prop-types';

class TextInput extends React.Component {
  constructor() {
    super();
    this.change = this.change.bind(this);
  }

  change(event) {
    this.props.callback(event.target.id, event.target.value);
  }

  render() {
    const { label, name, type, value, callback } = this.props;

    return(
      <div className="input-wrapper">
        <label htmlFor={name}>
          {label}
        </label>
        <input
          className="text-input"
          type={type}
          id={name}
          onChange={this.change}
        />
      </div>
    )
  };
}


TextInput.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  callback: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['password', 'email', 'text', 'number', 'tel']).isRequired,
};

export default TextInput;
