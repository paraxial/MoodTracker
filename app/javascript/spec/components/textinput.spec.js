import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TextInput from '../../diary/components/TextInput';

describe('Signup', () => {
  test('should match snapshot', () => {
    const renderer = new ShallowRenderer();
    const page = renderer
      .render(
        <TextInput
          label="password"
          name="password-input"
          type="password"
          value="some value"
          callback={() => (null)}
        />
      );

    expect(page).toMatchSnapshot();
  });

  test('should append focus class on focus', () => {

  });

  test('should remove focus class on blur when value is empty', () => {

  });

  test('should not remove focus class on blur when value is present', () => {

  });

  test('should call callback function with field and value on change', () => {

  });
});
