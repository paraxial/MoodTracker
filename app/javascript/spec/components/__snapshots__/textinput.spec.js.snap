// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<div
  className="input-wrapper"
>
  <label
    className="input-label"
    htmlFor="password-input"
  >
    password
  </label>
  <input
    className="text-input"
    id="password-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="password"
  />
</div>
`;
